tests/string_test.c

```diff
@@ -1,382 +1,63 @@
-
-    ASSERT_SUCCESS(aws_array_list_get_at(&output, &value, 0));
-    char *expected = "testa";
-
-    ASSERT_BIN_ARRAYS_EQUALS(expected, strlen(expected), value.ptr, value.len);
-
-    ASSERT_SUCCESS(aws_array_list_get_at(&output, &value, 1));
-    expected = "testb";
-    ASSERT_BIN_ARRAYS_EQUALS(expected, strlen(expected), value.ptr, value.len);
-
-    ASSERT_SUCCESS(aws_array_list_get_at(&output, &value, 2));
-    expected = "testc";
-    ASSERT_BIN_ARRAYS_EQUALS(expected, strlen(expected), value.ptr, value.len);
-
-    return 0;
-}
-
-AWS_TEST_CASE(test_char_split_output_too_small, test_char_split_output_too_small_fn)
-
-static int test_buffer_cat_fn(struct aws_allocator *allocator, void *ctx) {
-    struct aws_byte_buf str1 = aws_byte_buf_from_literal("testa");
-    struct aws_byte_buf str2 = aws_byte_buf_from_literal(";testb");
-    struct aws_byte_buf str3 = aws_byte_buf_from_literal(";testc");
-
-
-    const char expected[] = "testa;testb;testc";
-
-    struct aws_byte_buf destination;
-    ASSERT_SUCCESS(aws_byte_buf_init(allocator, &destination, str1.len + str2.len + str3.len + 10));
-    ASSERT_SUCCESS(aws_byte_buf_cat(&destination, 3, &str1, &str2, &str3));
-
-    ASSERT_INT_EQUALS(strlen(expected),destination.len);
-    ASSERT_INT_EQUALS(strlen(expected) + 10, destination.capacity);
-    ASSERT_BIN_ARRAYS_EQUALS(expected, strlen(expected), destination.buffer, destination.len);
-
-    aws_byte_buf_clean_up(&destination);
-
-    return 0;
-}
-
-AWS_TEST_CASE(test_buffer_cat, test_buffer_cat_fn)
-
-static int test_buffer_cat_dest_too_small_fn(struct aws_allocator *allocator, void *ctx) {
-    struct aws_byte_buf str1 = aws_byte_buf_from_literal("testa");
-    struct aws_byte_buf str2 = aws_byte_buf_from_literal(";testb");
-    struct aws_byte_buf str3 = aws_byte_buf_from_literal(";testc");
-
-    struct aws_byte_buf destination;
-    ASSERT_SUCCESS(aws_byte_buf_init(allocator, &destination, str1.len + str2.len));
-    ASSERT_INT_EQUALS(0, destination.len);
-    ASSERT_INT_EQUALS(str1.len + str2.len, destination.capacity);
+AWS_TEST_CASE(string_tests, string_tests_fn);
+static int string_tests_fn(struct aws_allocator *alloc, void *ctx) {
+    /* Test: static string creation from macro works. */
+    AWS_STATIC_STRING_FROM_LITERAL(test_string_1, "foofaraw");
+    ASSERT_NULL(test_string_1->allocator, "Static string should have no allocator.");
+    ASSERT_INT_EQUALS(test_string_1->len, 8, "Length should have been set correctly.");
+    ASSERT_BIN_ARRAYS_EQUALS(aws_string_bytes(test_string_1), test_string_1->len, "foofaraw", 8,
+                             "Data bytes should have been set correctly.");
 
-    ASSERT_ERROR(AWS_ERROR_DEST_COPY_TOO_SMALL, aws_byte_buf_cat(&destination, 3, &str1, &str2, &str3));
-
-    aws_byte_buf_clean_up(&destination);
-
-    return 0;
-}
+    /* Test: string creation works. */
+    const struct aws_string * test_string_2 = aws_string_from_c_str_new(alloc, "foofaraw", 8);
```

Comment:
The `c_str` part would, to me, imply a nul-terminated string, not a length delimited argument.

Reply:
I agree with this. As I mentioned above, this was not my naming convention. I have changed the naming convention to match what you want here.